sw/device/lib/testing/usb_testutils.h

```diff
@@ -12,6 +12,84 @@
 #include "sw/device/lib/dif/dif_usbdev.h"
 #include "usb_testutils_diags.h"
 
+// Result codes to rx/tx callback handlers
+typedef enum {
+  /**
+   * Successful completion.
+   */
+  kUsbTestutilsXfrResultOk = 0u,
+  /**
+   * Failed to transfer because of internal error,
+   * eg. buffer exhaustion.
+   */
+  kUsbTestutilsXfrResultFailed = 1u,
+  /**
+   * Link reset interrupted transfer.
+   */
+  kUsbTestutilsXfrResultLinkReset = 2u,
+  /**
+   * Canceled by suspend, endpoint removal or finalization.
+   */
+  kUsbTestutilsXfrResultCanceled = 3u,
+} usb_testutils_xfr_result_t;
+
+// Flags affecting the transfer of larger data buffers
+typedef enum {
+  kUsbTestutilsXfrMaxPacketMask = 0x7fu,  // Max packet size [0,0x40U]
+  /**
+   * Explicitly specify the maximum packet size; otherwise the device default
+   * of USBDEV_MAX_PACKET_SIZE shall be assumed.
+   */
+  kUsbTestutilsXfrMaxPacketSupplied = 0x100u,
```

Comment:
TL;DR: We can punt this for later, since there aren't any users yet.

This is not the end of the world, but it looks to me like we need information about the endpoint to be accessible at this layer. From an API perspective, it's awkward to associate this endpoint property with a struct that models one of its transfers. Instead, the transfer struct probably ought to have a pointer or ID that can be used to look up the required endpoint info.

From a caching perspective, though, perhaps it makes sense to have the max packet size close, especially if it's the only endpoint property we need. In that case, it'd probably be best left in its own field, marked as reserved for this specific layer.

The buffer handling layer of usb_testutils never needed to know the max packet size before--Since it only handled a single packet at a time, it could punt that up to the endpoint- or interface-specific code, which had the duty of forming the packet correctly. Now that you're adding a model of transfers to this layer, that changes.